File: swift/etl/Sources/Etl/Etl.swift

struct ETL {
   static func transform(_ oldScores: [Int: [String]]) -> [String: Int] {
      var newScores = [String: Int]()
      
      for score in oldScores.keys {
         for letter in oldScores[score]! {
            newScores[letter.lowercased()] = score
         }
      }
      return newScores
   }
}

File: go/etl/etl.go
// Package etl implements the etl exercise on exercism.io
package etl

import "strings"

// Transform converts old scrabble scores written in old scrabble system to use
// the new system
func Transform(input map[int][]string) map[string]int {
	result := make(map[string]int)
	for score, letters := range input {
		for _, letter := range letters {
			result[strings.ToLower(letter)] = score
		}
	}

	return result
}
